src/status_im/contexts/wallet/events.cljs

```diff
@@ -439,3 +439,32 @@
                 :type     :negative
                 :text     (i18n/label :t/provider-is-down {:chains chain-names})
                 :duration 10000}]]])})))
+
+(rf/reg-event-fx
+ :wallet/save-address
+ (fn [_ [{:keys [address name color-id on-success on-error chain-short-names ens is-test]
+         :or   {on-success        (fn [])
+                on-error          (fn [])
+                name              ""
+                ens               ""
+                is-test           false
+                chain-short-names "eth:"}}]]
+   (let [address-to-save {:address           address
+                          :name              name
+                          :color-id          color-id
+                          :ens               ens
+                          :is-test           is-test
+                          :chain-short-names chain-short-names}]
+     {:json-rpc/call
+      [{:method     "wakuext_upsertSavedAddress"
+        :params     [address-to-save]
+        :on-success on-success
+        :on-error   on-error}]})))
+
+(rf/reg-event-fx
+ :wallet/get-saved-addresses
+ (fn [_ [{:keys [on-success on-error]}]]
```

Comment:
usually these callbacks are called `on-success` and `on-fail`. Would be better to keep this more or less consistent wdyt?

Reply:
IMO `on-error` is not bad, seems we use it even more.

And I think it should be `on-failure` instead of `on-fail` :thinking:

Reply:
seems like you're right. Guess I was _lucky_ to mostly encounter the `on-fail` one